### PR TITLE
[W5.7][W17-2]Lin Deming

### PR DESCRIPTION
**Pull request by deming16**
Ready for review

### CODE REVIEW

**Comment by CrimsonJacket** [comment]
Brilliant Job @deming16 
Some considerations that you might want to take note of:
A common Software Development Model is Iterative & Incremental Development

> Iterative & Incremental Development
General idea is to develop a system through iterations (repeated cycles) and incrementally (in small portions of time). Through them team members or stakeholders can learn from their mistakes and apply that knowledge on the next iteration.

Maybe consider making multiple commits for the different additions
i.e 
Commit 1 : editted find command(case-insensitive)...
Commit 2 : include tests for case-insensitive...
Commit 3 : docs: find command(case insenstitive)...

**Comment by etjk** [comment]
Good job @deming16 
One other case that you may consider is to activate the case-insensitive search through a command instead of making the default search case-insensitive.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -79,14 +79,14 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
 
 [NOTE]
 ====
-The search is case sensitive, the order of the keywords does not matter, only the name is searched,
+The search is not case sensitive, the order of the keywords does not matter, only the name is searched,
```
Comment (amrut-prabhu) [inline docs/UserGuide.adoc:82]: Good job for updating the User Guide!

File: src/seedu/addressbook/commands/FindCommand.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.lang.*;
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/FindCommand.java:8]: Avoid batch imports.

You can change the IntelliJ settings to help prevent this.

File: src/seedu/addressbook/commands/FindCommand.java [hunk 3]
```diff
@@ -49,12 +50,28 @@ public CommandResult execute() {
...
+     * Converts all keywords to lowerCase.
+     *
+     * @param keywords for searching
+     * @return Set of keywords that has been converted to lower case
+     */
+    private Set<String> convertToLowerCase(Set<String> keywords) {
+        Set<String> keywordsInLowerCase = new HashSet<>();
+
+        for(String keyword : keywords) {
+          keywordsInLowerCase.add(keyword.toLowerCase());
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/FindCommand.java:71]: Indentation should be 4 spaces!

File: test/expected.txt [hunk 4]
```diff
@@ -14,7 +14,7 @@
 || Example: delete 1
 || Clears address book permanently.
 || Example: clear
-|| find: Finds all persons whose names contain any of the specified keywords (case-sensitive) and displays them as a list with index numbers.
+|| find: Finds all persons whose names contain any of the specified keywords (case-insensitive) and displays them as a list with index numbers.
```
Comment (amrut-prabhu) [inline test/expected.txt:17]: Good job for updating I/O tests

File: test/java/seedu/addressbook/commands/FindCommandTest.java [hunk 5]
```diff
@@ -25,8 +25,8 @@ public void execute() throws IllegalValueException {
         //same word, same case: matched
         assertFindCommandBehavior(new String[]{"Amy"}, Arrays.asList(td.amy));
 
-        //same word, different case: not matched
-        assertFindCommandBehavior(new String[]{"aMy"}, Collections.emptyList());
+        //same word, different case: matched
```
Comment (amrut-prabhu) [inline test/java/seedu/addressbook/commands/FindCommandTest.java:28]: Good work for updating the JUnit test!

**Review by amrut-prabhu — APPROVED** [review]
Good job for the LO @deming16 
Please close the PR after reading my comments

**Pull request closed without merging**